evervault_test.go

```diff
@@ -391,35 +380,23 @@ func testFuncHandler(writer http.ResponseWriter, reader *http.Request, mockRespo
 	writer.WriteHeader(http.StatusOK)
 	writer.Header().Set("Content-Type", "application/json")
 
-	// statusResponse, statusOk := mockResponse["status"].(string)
-	// if !statusOk {
-	// 	statusResponse = ""
-	// }
-
-	// idResponse, ok := mockResponse["id"].(string)
-	// if !ok {
-	// 	idResponse = ""
-	// }
-
-	// resultResponse, ok := mockResponse["result"].(map[string]any)
-	// if !ok {
-	// 	resultResponse = map[string]any{}
-	// }
-
-	// responseBody := evervault.FunctionRunResponse{
-	// 	Status: statusResponse,
-	// 	Id: idResponse,
-	// 	Result: resultResponse,
-	// }
-	// if err := json.NewEncoder(writer).Encode(responseBody); err != nil {
-	// 	log.Printf("error encoding json: %s", err)
-	// }
-
-	writer.Write([]byte(mockResponse))
+	// Handle mockResponse based on its type
+	switch v := mockResponse.(type) {
+	case []byte:
+		writer.Write(v) // mockResponse is already []byte
+	case string:
+		writer.Write([]byte(v)) // Convert string to []byte
```

Comment:
should string go through the jsonEncoder to ensure it's consistent with the API?

Reply:
Yeah good shout, updated